src/main/jbake/content/blog/2017/01/13/events-aggregation-extension.adoc

```diff
@@ -0,0 +1,189 @@
+          "type": "EXTERNAL",
+          "alerterId":"EventsAggregation",
+          "dataId": "marketing",
+          "expression": "event:groupBy(context.accountId):window(time,10s):having(count > 2)"
+        }
+      ]
+    }
+  ],
+  "actions":[
+    {
+      "actionPlugin": "email",
+      "actionId": "notify-to-marketing",
+      "properties": {
+        "to": "marketing@hawkular.org"
+      }
+    }
+  ]
+}
+
+
+The expression used can be described as follows:
+
+[source]
+groupBy(context.accountId)      Group window events by context "accountId" field
+window(time,10s)                Define a sliding time window of 10 seconds
+having(count > 2)               Define an expression on the grouped events
+
+In other words, this condition will be true, each time that there are more than two events with the same _accountId_ for a 10 seconds window.
+
+In a similar way, we can describe the *Fraud* scenario previously described with the expression:
+
+[source,json]
+"event:groupBy(tags.accountId):window(time,10s):having(count > 1, count.tags.location > 1)"
+
+Where
+
+[source]
+groupBy(context.accountId)                    Group window events by context "accountId" field
+window(time,10s)                              Define a sliding time window of 10 seconds
+having(count > 1, count.tags.location > 1)    Define an expression on the grouped events
+
+This condition will be true when there are more than 1 events with more than one _location_ tag, so detecting when
+events for the same _accountId_ happens from different places.
+
+The two previous expressions groups all events of the timing window.
+We might have scenarios where only specific events should be grouped.
+
+For these cases we can add filters into the expressions like in the following example:
+
+[source,json]
+"event:groupBy(tags.traceId):filter((category == \"Credit Check\" && text == \"Exceptionally Good\") || (category == \"Stock Check\" && text == \"Out of Stock\")):having(count > 1, count.tags.accountId == 1)"
+
+This expression will group events filtered by a expression
+
+[source]
+filter(
+    (category == \"Credit Check\" && text == \"Exceptionally Good\") ||
+    (category == \"Stock Check\" && text == \"Out of Stock\")
+)
+
+Note that this expression doesn't define an explicit sliding time window, so it will use a default expiration window.
+
+=== Use cases
+
+Stream Processing and Polling Processing might be used for similar scenarios.
+
+The EventsAggregation Extension groups Events in memory so it is designed for real time scenarios with relatively short sliding windows.
+
+== Conclusion
+
+EventsAggregation Extension is a powerful addition to Hawkular Alerting that will extend the scenarios and type of behaviours that can be detected.
+
+In our future work we will enhance this extension covering more use cases (potential aggregation of data).
+
+We hope this small intro could help to show the new CEP capabilities for Hawkular Alerting.
```

Comment:
proposed: We hope this short introduction helps show how the EventsAggregation Extension provides powerful new CEP capabilities for Hawkular Alerting.